export_pool_opt_in_contacts.py

```diff
@@ -0,0 +1,108 @@
+                                                 "Pool at the end of the project")
+
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file",
+                        help="Path to the pipeline configuration json file")
+    parser.add_argument("traced_data_paths", metavar="traced-data-paths", nargs="+",
+                        help="Paths to the traced data files (either messages or individuals) to extract phone "
+                             "numbers from")
+    parser.add_argument("csv_output_file_path", metavar="csv-output-file-path",
+                        help="Path to a CSV file to write the contacts to. "
+                             "Exported file is in a format suitable for direct upload to Rapid Pro")
+
+    args = parser.parse_args()
+
+    google_cloud_credentials_file_path = args.google_cloud_credentials_file_path
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    traced_data_paths = args.traced_data_paths
+    csv_output_file_path = args.csv_output_file_path
+
+    sys.setrecursionlimit(10000)
+
+    log.info("Loading Pipeline Configuration File...")
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+    Logger.set_project_name(pipeline_configuration.pipeline_name)
+    log.debug(f"Pipeline name is {pipeline_configuration.pipeline_name}")
+
+    log.info("Downloading Firestore UUID Table credentials...")
+    firestore_uuid_table_credentials = json.loads(google_cloud_utils.download_blob_to_string(
+        google_cloud_credentials_file_path,
+        pipeline_configuration.phone_number_uuid_table.firebase_credentials_file_url
+    ))
+
+    phone_number_uuid_table = FirestoreUuidTable(
+        pipeline_configuration.phone_number_uuid_table.table_name,
+        firestore_uuid_table_credentials,
+        "avf-phone-uuid-"
+    )
+    log.info("Initialised the Firestore UUID table")
+
+    uuids = set()
+    skipped = 0
+    for path in traced_data_paths:
+        # Load the traced data
+        log.info(f"Loading previous traced data from file '{path}'...")
+        with open(path) as f:
+            data = TracedDataJsonIO.import_jsonl_to_traced_data_iterable(f)
+        log.info(f"Loaded {len(data)} traced data objects")
+
+        # Filter for participants that opted-in via the close-out.
+        for td in data:
+            if td["consent_withdrawn"] == Codes.TRUE:
+                continue
+
+            plan = PipelineConfiguration.RQA_CODING_PLANS[8]
+            assert plan.raw_field == "s01_close_out_raw"
```

Comment:
move plan.raw_field to args so that we can easily copy it into other pipelines

Reply:
Good suggestion :) I've removed this and replaced with a new program argument, "opt_in_dataset_name", which specifies the dataset_name of the coding plan to search for opt-ins, so this should be easier to reuse in future.